src/main/java/io/vertx/core/json/Json.java

```diff
@@ -72,12 +74,26 @@ public static String encodePrettily(Object obj) throws EncodeException {
   }
 
   public static <T> T decodeValue(String str, Class<T> clazz) throws DecodeException {
+    return decodeValue(str, clazz, true);
+  }
+
+  public static <T> T decodeValue(String str, Class<T> clazz, boolean failOnUnknownProperties) throws DecodeException {
+    ObjectReader reader = mapper.readerFor(clazz);
+
+    if (failOnUnknownProperties) {
+      reader = reader.with(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    } else {
+      reader = reader.without(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    }
+
```

Comment:
Okay. I modified so that the non-default reader is only created when calling for non-default behavior. Yes, this does rely on the default mapper being configured to fail on unknown properties, but that was the presumption before as well.